tests/cpp/test_serializer.cc

```diff
@@ -189,10 +189,10 @@ void PyBufferInterfaceRoundTrip_TreeDepth2() {
     tree->SetCategoricalTestNode(1, 0, {0, 1}, true, 3, 4);
     tree->SetCategoricalTestNode(2, 1, {0}, true, 5, 6);
     tree->SetRootNode(0);
-    tree->SetLeafNode(3, frontend::Value::Create<LeafOutputType>(3));
-    tree->SetLeafNode(4, frontend::Value::Create<LeafOutputType>(1));
-    tree->SetLeafNode(5, frontend::Value::Create<LeafOutputType>(4));
-    tree->SetLeafNode(6, frontend::Value::Create<LeafOutputType>(2));
+    tree->SetLeafNode(3, frontend::Value::Create<LeafOutputType>(tree_id + 3));
+    tree->SetLeafNode(4, frontend::Value::Create<LeafOutputType>(tree_id + 1));
+    tree->SetLeafNode(5, frontend::Value::Create<LeafOutputType>(tree_id + 4));
+    tree->SetLeafNode(6, frontend::Value::Create<LeafOutputType>(tree_id + 2));
```

Comment:
I crafted the test case so that all the trees will have different leaf outputs. The bug being addressed causes all trees to be identical.